lms/static/js/student_account/components/StudentAccountDeletion.jsx

```diff
@@ -45,10 +61,26 @@ export class StudentAccountDeletion extends React.Component {
         <Button
           id="delete-account-btn"
           className={['btn-outline-primary']}
+          disabled={this.state.socialAuthConnected}
           label={gettext('Delete My Account')}
           inputRef={(input) => { this.modalTrigger = input; }}
           onClick={this.loadDeletionModal}
         />
+        {this.state.socialAuthConnected && <StatusAlert
```

Comment:
You could update line 34 to be ```const { deletionModalOpen, socialAuthConnected } = this.state;``` and then you can just use the constant ```socialAuthConnected``` throughout.